raiden-ts/src/messages/utils.ts

```diff
@@ -54,61 +84,40 @@ export function createMessageHash(message: EnvelopeMessage): Hash {
     case MessageType.LOCKED_TRANSFER:
     case MessageType.REFUND_TRANSFER:
       // hash of packed representation of the whole message
-      return keccak256(
-        concat([
-          encode(CMDIDs[message.type], 1),
-          encode(0, 3), // pad(3)
-          encode(message.nonce, 8),
-          encode(message.chain_id, 32),
-          encode(message.message_identifier, 8),
-          encode(message.payment_identifier, 8),
-          encode(message.lock.expiration, 32),
-          encode(message.token_network_address, 20),
-          encode(message.token, 20),
-          encode(message.channel_identifier, 32),
-          encode(message.recipient, 20),
-          encode(message.target, 20),
-          encode(message.initiator, 20),
-          encode(message.locksroot, 32),
-          encode(message.lock.secrethash, 32),
-          encode(message.transferred_amount, 32),
-          encode(message.locked_amount, 32),
-          encode(message.lock.amount, 32),
-          encode(message.fee, 32),
-        ]),
-      ) as Hash;
+      const packed = concat([
+        encode(CMDIDs[message.type], 1),
+        encode(message.message_identifier, 8),
+        encode(message.payment_identifier, 8),
+        encode(message.lock.expiration, 32),
+        encode(message.token, 20),
+        encode(message.recipient, 20),
+        encode(message.target, 20),
+        encode(message.initiator, 20),
+        encode(message.lock.secrethash, 32),
+        encode(message.lock.amount, 32),
+        encode(message.fee, 32),
+      ]);
+      const hashable =
+        message.type === MessageType.LOCKED_TRANSFER
+          ? concat([packed, createMetadataHash(message.metadata)])
+          : packed;
+      return keccak256(hashable) as Hash;
     case MessageType.UNLOCK:
       return keccak256(
         concat([
           encode(CMDIDs[message.type], 1),
-          encode(0, 3),
```

Comment:
Whoah, they really stripped the `additional_hash` for these messages, didn't they? I hope it doesn't miss important data for the integrity check. This doesn't affect the smartcontract at all, as soon as both clients compose this `additional_hash` exactly the same, so tests which compare our packing with output of python client should be enough. Thank you for going over it

Reply:
They removed quite a few properties from the hashing yes.